src/seedu/addressbook/commands/Command.java

```diff
@@ -63,6 +63,14 @@ protected ReadOnlyPerson getTargetPerson() throws IndexOutOfBoundsException {
         return relevantPersons.get(getTargetIndex() - DISPLAYED_INDEX_OFFSET);
     }
 
+    /**
+     * Determines if command is mutating the data
+     * @return true if is mutating and false if not
+     */
+    public boolean isMutating() {
+        return false;
+    }
+
```

Comment:
Ah remember that we talked about this in class, and how it may not be a good idea to do this? :(

Having a default implementation for the `isMutating` method helps to reduce the amount of code. However, some may argue that choosing `false` (or `true`) as the default is arbitrary and hurts the intuitiveness of the code. Making the method `abstract` forces all child classes to declare whether it is a mutating command or not, rather than depend on an arbitrary decision made in the parent class.